app/assets/javascripts/validation.js

```diff
@@ -10,69 +10,183 @@ $(document).ready(function () {
-        minlength: "Password should be minimum 8 characters long"
-      },
-      email: {
-        required: "Please enter email",
-        email: "Invalid email address"
-      },
-      password2: {
-        required: "Please enter password",
-        minlength: "Password should be minimum 8 characters long",
-        equalTo: "Passwords doesn't match"
-      }
-    },
-    submitHandler: function(form) {
-      form.submit();
+$(document).ready(function () {
+  var signUpForm = document.querySelector('#create-form');
+
+  if (!signUpForm) return;
+
+  var validationTracker = {};
+  var submitBtn = document.querySelector("#create-form [type='submit']");
+  var usernameElement = document.querySelector("[name='user[username]']");
+  var emailElement = document.querySelector("[name='user[email]']");
+  var passwordElement = document.querySelector("[name='user[password]']");
+  var confirmPasswordElement = document.querySelector(
+    "[name='user[password_confirmation]']"
+  );
+
+  isFormValid();
+
+  // Every time user types something, it triggers corresponding event listener
+  usernameElement.addEventListener('input', validateUsername);
+  emailElement.addEventListener('input', validateEmail);
+  passwordElement.addEventListener('input', validatePassword);
+  confirmPasswordElement.addEventListener('input', validateConfirmPassword);
+
+  function validateUsername(e) {
+    var username = e.target.value;
+
+    if (username.length < 3) {
+      restoreOriginalStyle(this);
+    } else {
+      $.get('/api/srch/profiles?query=' + username, function(data) {
+        if (data.items) {
+          $.map(data.items, function(userData) {
+            if (userData.doc_title === username) {
+              updateUI(usernameElement, false, 'Username already exists');
+            } else {
+              updateUI(usernameElement, true);
+            }
+          });
+        } else {
+          updateUI(usernameElement, true);
+        }
+      });
     }
-  });
+  }
+
+  function validateEmail(e) {
+    var email = e.target.value;
+    var regexp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    var isValid = regexp.test(email);
+
+    updateUI(emailElement, isValid, 'Invalid email');
+  }
+
+  function validatePassword(e) {
+    var password = e.target.value;
+
+    if (!isPasswordValid(password)) {
+      updateUI(this, false, 'Minimum 8 characters including 1 number');
+      return;
+    }
+
+    if (password === confirmPasswordElement.value) {
+      updateUI(confirmPasswordElement, true);
+    }
+
+    updateUI(passwordElement, true);
+  }
+
+  function validateConfirmPassword(e) {
+    var password = passwordElement.value;
+    var confirmPassword = e.target.value;
+
+    if (!isPasswordValid(confirmPassword)) {
+      updateUI(this, false, 'Minimum 8 characters including 1 number');
+      return;
+    }
+
```

Comment:
is IF condition needed here as below if we automatically check it?

Reply:
This condition is necessary. It will first check if the password contains 8 characters including one number. If that is true, then it will check if both passwords are equal and style element accordingly.

Without this "if" statement it would just display "Passwords must be equal", and it won't tell anything about whether the password is technically valid(8 chars + 1 num).

I hope it's clear now.

Reply:
Mainly the confirm password is for checking whether the password entered before matches  with the password retyped 
Password format validation is done only when it first typed by the user. That why i was saying.@

Reply:
@keshav234156 thanks for showing interest in this feature.
It is debatable, I think it results in nicer UI/UX but we can remove it if you think that's right thing to do.

Reply:
@SidharthBansal your views?

Reply:
I agree with keshav

Reply:
Alright, I'll change it :+1: